### PR TITLE
SPU LLVM: Add accurate double-precision FMA support

### PR DESCRIPTION
**Pull request by elad335**
Note: the setting's value of "PPU Accurate FMA" will reset.. it has been renamed to "LLVM Accurate FMA".

Also fix a corner case of SPU FNMS in LLVM decoders.

### CODE REVIEW

File: rpcs3/rpcs3qt/settings_dialog.ui [hunk 1]
```diff
@@ -1718,7 +1718,7 @@
                </widget>
               </item>
             <item>
-             <widget class="QCheckBox" name="accuratePPUfma">
+             <widget class="QCheckBox" name="accurateLLVMfma">
               <property name="text">
                <string>Accurate PPU FMA</string>
```
Comment (illusion0001) [inline rpcs3/rpcs3qt/settings_dialog.ui:1723]: Still display PPU here

**Comment by Xcedf** [comment]
@elad335 Don't you know, do these double-precise instructions used anywhere?

**Comment by Nekotekina** [comment]
It can have less confusing name, like "Accurate DFMA"

**Comment by elad335** [comment]
Fixed and added optimization for Negate+Substruct type FMA instructions (both PPU/SPU).

**Comment by Nekotekina** [comment]
Oh, that's a lot of conflicts.

**Comment by elad335** [comment]
Conflicts solved, I tested to confrim the gui option still works and it does.

**Pull request merged**